pilot/pkg/model/context.go

```diff
@@ -208,10 +210,10 @@ var DefaultXdsLogDetails = XdsLogDetails{}
 // Note: any errors returned will completely close the XDS stream. Use with caution; typically and empty
 // or no response is preferred.
 type XdsResourceGenerator interface {
-	Generate(proxy *Proxy, push *PushContext, w *WatchedResource, updates *PushRequest) (Resources, XdsLogDetails, error)
-
-	// GenerateDeltas returns the changed and removed resources, along with whether or not delta was actually used.
-	GenerateDeltas(proxy *Proxy, push *PushContext, updates *PushRequest, w *WatchedResource) (Resources, []string, XdsLogDetails, bool, error)
+	// Generate builds the xds resources for given proxy. Resources contain the newly added/updated resources.
+	// DeletedResources typically used in incremental Xds protocols contain the resources that have been deleted.
+	// It also indicates whether the response is delta or SoTW.
```

Comment:
can we say it is delta if no deleted resources? I understand delta xds can also contain no deleted resources, maybe i am wrong

Reply:
Right now we can not say if deleted resources are empty that is because we do partial delta processing in CDS i.e. we do only delta logic when all updates are of either Service or DR. So that explicit flag is needed.